Lib/test/test_pathlib/test_pathlib.py

```diff
@@ -1102,16 +1102,18 @@ def test_is_socket_true(self):
         self.assertIs(self.cls(self.base, 'mysock\udfff').is_socket(), False)
         self.assertIs(self.cls(self.base, 'mysock\x00').is_socket(), False)
 
+    @unittest.skipIf(
+        is_emscripten or is_wasi,
+        "devnull is not implemented on Emscripten/WASI."
+    )
```

Comment:
What does it mean? `P.exists()` returns false?

Reply:
I wonder whether the existing code
```py
if not P.exists():
    self.skipTest(...)
```
handles it better. If someday Emscripten/WASI will get devnull, or new platform without devnull will be added...

Reply:
I took a look and found that [Emscripten](https://emscripten.org/docs/api_reference/Filesystem-API.html) should have the `/dev/null` device. Maybe it's caused by the implementation, or it's an issue with `is_char_device`.

I think the approach you provide should be okay. I'll try it.